### PR TITLE
Test1

### PR DESCRIPTION
**Pull request by revmag**
All the changes made to the main refusal_dir repo

### CODE REVIEW

File: pipeline/submodules/select_direction.py [hunk 1]
```diff
@@ -114,6 +114,137 @@ def filter_fn(refusal_score, steering_score, kl_div_score, layer, n_layer, kl_th
...
+            })
+
+            refusal_score = ablation_refusal_scores[source_pos, source_layer].item()
+            steering_score = steering_refusal_scores[source_pos, source_layer].item()
+            kl_div_score = ablation_kl_div_scores[source_pos, source_layer].item()
+
+    with open(f"{artifact_dir}/direction_evaluations.json", 'w') as f:
+        json.dump(json_output_all_scores, f, indent=4)
+
+    return -2, 10, direction_to_add
```
Comment (taufeeque9) [inline pipeline/submodules/select_direction.py:246]: This is a useless return statement. You can just remove it or return something useful.

File: pipeline/submodules/select_direction.py [hunk 2]
```diff
@@ -114,6 +114,137 @@ def filter_fn(refusal_score, steering_score, kl_div_score, layer, n_layer, kl_th
         return True
     return False
 
+def select_direction_2(
```
Comment (taufeeque9) [inline pipeline/submodules/select_direction.py:117]:
```suggestion
def eval_refusal_dir_across_layers(
```

**Review by taufeeque9 — APPROVED** [review]
Looks good. Suggested some changes that you can incorporate. One general tips: Assign better names to variables and function names. Don't be afraid to use long names.

File: pipeline/submodules/select_direction.py [hunk 3]
```diff
@@ -114,6 +114,137 @@ def filter_fn(refusal_score, steering_score, kl_div_score, layer, n_layer, kl_th
...
+    if not os.path.exists(artifact_dir):
+        os.makedirs(artifact_dir)
+
+    d_model = candidate_directions.shape
+    n_layer = 18
+        
+    direction_to_add = candidate_directions
+
+    baseline_refusal_scores_harmful = get_refusal_scores(model_base.model, harmful_instructions, model_base.tokenize_instructions_fn, model_base.refusal_toks, fwd_hooks=[], batch_size=batch_size)
+    baseline_refusal_scores_harmless = get_refusal_scores(model_base.model, harmless_instructions, model_base.tokenize_instructions_fn, model_base.refusal_toks, fwd_hooks=[], batch_size=batch_size)
```
Comment (cybershiptrooper) [inline pipeline/submodules/select_direction.py:137]: Can we rename refusal_scores to be consistent with the paper (afair, it is calculating the refusal_metric and not the score?) It's quite confusing :(

File: pipeline/submodules/select_direction.py [hunk 4]
```diff
@@ -114,6 +114,137 @@ def filter_fn(refusal_score, steering_score, kl_div_score, layer, n_layer, kl_th
...
+            refusal_vector = direction_to_add
+            coeff = torch.tensor(1.0)
+
+            fwd_pre_hooks = [(model_base.model_block_modules[source_layer], get_activation_addition_input_pre_hook(vector=refusal_vector, coeff=coeff))]
+            fwd_hooks = []
+
+            refusal_scores = get_refusal_scores(model_base.model, harmless_instructions, model_base.tokenize_instructions_fn, model_base.refusal_toks, fwd_pre_hooks=fwd_pre_hooks, fwd_hooks=fwd_hooks, batch_size=batch_size)
+            steering_refusal_scores[source_pos, source_layer] = refusal_scores.mean().item()
+
+    plot_refusal_scores(
```
Comment (cybershiptrooper) [inline pipeline/submodules/select_direction.py:197]: Would be great to have the error bars as well...

**Review by cybershiptrooper — APPROVED** [review]

**Pull request merged**